src/metabase/lib/metadata.cljc

```diff
@@ -152,12 +152,17 @@
    card-id               :- ::lib.schema.id/card]
   (lib.metadata.protocols/card (->metadata-provider metadata-providerable) card-id))
 
+(mu/defn cached-cards #_:- #_[:maybe [:sequential ::lib.schema.metadata/card]]
+  "Get metadata for a Card, aka Saved Question, with `card-id`, if it can be found."
+  [metadata-providerable :- MetadataProviderable]
+  (lib.metadata.protocols/cached-metadata (->metadata-provider metadata-providerable) :metadata/card))
```

Comment:
I think we should just expose `cached-metadata` here (basically just have a version that calls `->metadata-provider` on its `metadata-providerable` arg) that's a lot more general and potentially useful for more than just this one very specific use case.